datadog_lambda/tracing.py

```diff
@@ -226,6 +252,9 @@ def extract_context_from_sqs_or_sns_event_or_context(event, lambda_context):
 
     try:
         first_record = event.get("Records")[0]
```

Comment:
You should define `arn = None` or `arn = ""` above as you did with `is_sqs`.

In the case where this line fails, `arn` will not yet be defined leading to an exception being raised during the exception handling.